plugin/src/test/groovy/org/gradle/testretry/testframework/JUnit5FuncTest.groovy

```diff
@@ -718,6 +719,63 @@ class JUnit5FuncTest extends AbstractFrameworkFuncTest {
         gradleVersion << GRADLE_VERSIONS_UNDER_TEST
     }
 
+    def "supports dynamic tests (gradle version #gradleVersion)"() {
+        given:
+        buildFile << """
+            test {
+                retry {
+                    maxRetries = 1
+                }
+            }
+        """
+
+        writeJavaTestSource """
+            package acme;
+
+            import org.junit.jupiter.api.*;
+            import java.util.stream.Stream;
+
+            class MyTest {
+                @TestFactory
+                DynamicContainer dynamicContainerTest() {
+                    return DynamicContainer.dynamicContainer("container", Stream.of(
+                        DynamicTest.dynamicTest("test name 1", () -> {
+                            ${flakyAssert()}
+                        })
+                    ));
+                }
+            }
+        """
+
+        when:
+        def result = gradleRunner(gradleVersion).build()
+
+        then:
+        def gv = GradleVersion.version(gradleVersion)
+        def testname
+        if (gv >= GradleVersion.version("8.8")) {
```

Comment:
❔ Can we extract it into a method like `dynamicTestName(version, name)` that would just return the string we should assert on?